src/components/Documentation/RightPanel/index.tsx

```diff
@@ -34,6 +35,31 @@ const RightPanel: React.FC<IRightPanelProps> = ({
   const [currentHeadingSlug, setCurrentHeadingSlug] = useState<string | null>(
     null
   )
+  const generateHeadingId = (slug: string): string => `links-${slug}`
+  const tableOfContentsRefs: {
+    [key: string]: React.RefObject<HTMLDivElement>
+  } = headings
+    .map(({ slug }) => ({
+      id: generateHeadingId(slug),
+      ref: createRef()
+    }))
+    .reduce((acc, curr) => {
+      return {
+        ...acc,
+        [curr.id]: curr.ref
+      }
+    }, {})
+
+  const scrollSidebar = (): void => {
+    if (currentHeadingSlug && tableOfContentsRefs) {
+      const target =
+        tableOfContentsRefs[generateHeadingId(currentHeadingSlug)].current
+      if (target) {
+        ;(target.parentNode as HTMLDivElement).scrollTop =
+          target.offsetTop - (target.parentNode as HTMLDivElement).offsetTop
```

Comment:
Using a `<HTMLDivElement>.scrollTop` based solution lets you control the scroll isolated to a div, whereas the alternative `<HTMLDivElement>. scrollIntoView ()` affected multiple scroll divs.